integrations/terraform-cloud/main.py

```diff
@@ -0,0 +1,115 @@
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+class ObjectKind(StrEnum):
+    WORKSPACE = "workspace"
+    RUN = "run"
+    STATE_VERSION_OUTPUT = "state-version-output"
+    STATE_VERSION = "state-version"
+
+
+def init_terraform_client() -> TerraformClient:
+    """
+    Intialize Terraform Client
+    """
+    config = ocean.integration_config
+    
+    terraform_client = TerraformClient(
+                    config["terraform_host"],
+                    config["terraform_token"],
+                    )
+
+    return terraform_client
+
+
+async def setup_application()->Any:
+        
+    app_host = ocean.integration_config.get("app_host")
+
+    if not app_host:
+        logger.warning(
+            "No app host provided, skipping webhook creation. "
+            "Without setting up the webhook, the integration will not export live changes from Terraform"
+        )
+        return
+
+    terraform_client = init_terraform_client()
+    await terraform_client.create_workspace_webhook(app_host=app_host)
+
+@ocean.router.post("/webhook")
+async def handle_webhook_request(data: dict[str, Any]) -> dict[str, Any]:
+    terraform_client = init_terraform_client()
+
+    run_id = data["run_id"]
+    logger.info(f"Processing Terraform run event for run: {run_id}")
+
+    workspace_id = data['workspace_id']
+    logger.info(f"Processing Terraform run event for workspace: {workspace_id}")
+
+    run, workspace = await gather(
+        terraform_client.get_single_run(run_id),
+        terraform_client.get_single_workspace(workspace_id)
+    )
+
+    await gather(
+        ocean.register_raw(ObjectKind.RUN, [run]),
+        ocean.register_raw(ObjectKind.WORKSPACE, [workspace])
+    )
+
+    logger.info("Terraform webhook event processed")
+    return {"ok": True}
+
+
+@ocean.on_resync(ObjectKind.WORKSPACE)
+async def resync_workspaces(kind: str) -> list[dict[Any, Any]]:
+    terraform_client = init_terraform_client()
+
+    async for workspace in terraform_client.get_paginated_workspaces():
+        logger.info(f"Received {len(workspace)} batch workspaces")
+        yield workspace
+
+
+@ocean.on_resync(ObjectKind.RUN)
+async def resync_runs(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    terraform_client = init_terraform_client()
+
+    async for workspaces in terraform_client.get_paginated_workspaces():
+        logger.info(f"Received {len(workspaces)} batch runs")
+        for workspace in workspaces:
+            async for runs in terraform_client.get_paginated_runs_for_workspace(workspace['id']):
+                yield runs
+
+
+@ocean.on_resync(ObjectKind.STATE_VERSION_OUTPUT)
+async def resync_runs(kind:str)-> ASYNC_GENERATOR_RESYNC_TYPE:
+    terraform_client = init_terraform_client()
+
+    async for state_versions_output in terraform_client.get_state_version_outputs_for_workspace():
+            logger.info(f"Received {len(state_versions_output)} batch state version outputs")
+            yield state_versions_output
```

Comment:
you sure we need a whole kind for this?

outputs are always connected to the state version and im pretty sure i saw a way to get it with the state version request

Reply:
Without state version output being a kind, we can get the id for a state version output from a state version, but the state version output data itself isn't inclusive. Please let me know if this is fine,
Here is the output of a state version request

{
    "data": [
        {
            "id": "sv-wZEoyjPg1KYsjYZg",
            "type": "state-versions",
            "attributes": {
                "created-at": "2023-12-20T07:08:14.113Z",
                "size": 8554,
                "hosted-state-download-url": "https://app.terraform.io/api/state-versions/sv-wZEoyjPg1KYsjYZg/hosted_state",
                "hosted-json-state-download-url": "https://app.terraform.io/api/state-versions/sv-wZEoyjPg1KYsjYZg/hosted_json_state",
                "modules": {
                    "root": {
                        "random_string": 1,
                        "aws_db_instance": 1,
                        "aws_db_subnet_group": 1,
                        "data.aws_availability_zones": 1
                    },
                    "root.vpc": {
                        "aws_eip": 2,
                        "aws_vpc": 1,
                        "aws_route": 3,
                        "aws_subnet": 4,
                        "aws_nat_gateway": 2,
                        "aws_route_table": 3,
                        "aws_internet_gateway": 1,
                        "aws_route_table_association": 4
                    },
                    "root.elb-http.elb": {
                        "aws_elb": 1
                    },
                    "root.ec2-instances": {
                        "aws_instance": 4,
                        "data.aws_ami": 1
                    },
                    "root.lb-security-group.sg": {
                        "aws_security_group": 1,
                        "aws_security_group_rule": 6
                    },
                    "root.app-security-group.sg": {
                        "aws_security_group": 1,
                        "aws_security_group_rule": 6
                    },
                    "root.elb-http.elb-attachment": {
                        "aws_elb_attachment": 4
                    }
                },
                "providers": {
                    "provider[\"registry.terraform.io/hashicorp/aws\"]": {
                        "aws_eip": 2,
                        "aws_elb": 1,
                        "aws_vpc": 1,
                        "aws_route": 3,
                        "aws_subnet": 4,
                        "aws_instance": 4,
                        "data.aws_ami": 1,
                        "aws_db_instance": 1,
                        "aws_nat_gateway": 2,
                        "aws_route_table": 3,
                        "aws_elb_attachment": 4,
                        "aws_security_group": 2,
                        "aws_db_subnet_group": 1,
                        "aws_internet_gateway": 1,
                        "aws_security_group_rule": 12,
                        "aws_route_table_association": 4,
                        "data.aws_availability_zones": 1
                    },
                    "provider[\"registry.terraform.io/hashicorp/random\"]": {
                        "random_string": 1
                    }
                },
                "resources": [
                    {
                        "name": "available",
                        "type": "data.aws_availability_zones",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "database",
                        "type": "aws_db_instance",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_db_subnet_group",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "lb_id",
                        "type": "random_string",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/random\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this_name_prefix",
                        "type": "aws_security_group",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "egress_rules",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_rules",
                        "type": "aws_security_group_rule",
                        "count": 4,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_with_self",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "amazon_linux",
                        "type": "data.aws_ami",
                        "count": 1,
                        "module": "root.ec2_instances",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "app",
                        "type": "aws_instance",
                        "count": 4,
                        "module": "root.ec2_instances",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_elb",
                        "count": 1,
                        "module": "root.elb_http.elb",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_elb_attachment",
                        "count": 4,
                        "module": "root.elb_http.elb_attachment",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this_name_prefix",
                        "type": "aws_security_group",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "egress_rules",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_rules",
                        "type": "aws_security_group_rule",
                        "count": 4,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_with_self",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "nat",
                        "type": "aws_eip",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_internet_gateway",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_nat_gateway",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private_nat_gateway",
                        "type": "aws_route",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public_internet_gateway",
                        "type": "aws_route",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_route_table",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_route_table",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_route_table_association",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_route_table_association",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_subnet",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_subnet",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_vpc",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    }
                ],
                "resources-processed": true,
                "serial": 3,
                "state-version": 4,
                "status": "finalized",
                "terraform-version": "1.6.6",
                "vcs-commit-url": null,
                "vcs-commit-sha": null
            },
            "relationships": {
                "run": {
                    "data": {
                        "id": "run-9XBzSuHzfgwTZNak",
                        "type": "runs"
                    }
                },
                "rollback-state-version": {
                    "data": null
                },
                "created-by": {
                    "data": {
                        "id": "user-VGzYMVaTX6nVYZ3U",
                        "type": "users"
                    },
                    "links": {
                        "self": "/api/v2/users/user-VGzYMVaTX6nVYZ3U",
                        "related": "/api/v2/runs/run-9XBzSuHzfgwTZNak/created-by"
                    }
                },
                "workspace": {
                    "data": {
                        "id": "ws-pN259iR1J3cW8ivr",
                        "type": "workspaces"
                    }
                },
                "outputs": {
                    "data": [
                        {
                            "id": "wsout-uJCPifEGM1fC5UF1",
                            "type": "state-version-outputs"
                        },
                        {
                            "id": "wsout-SD3QFXBkcK41c8Vq",
                            "type": "state-version-outputs"
                        },
                        {
                            "id": "wsout-ryDfZJ4dwxuqR5NU",
                            "type": "state-version-outputs"
                        },
                        {
                            "id": "wsout-5vnxsMuMaorwojVL",
                            "type": "state-version-outputs"
                        },
                        {
                            "id": "wsout-cP7abYGPeXpgJfrt",
                            "type": "state-version-outputs"
                        }
                    ],
                    "links": {
                        "related": "/api/v2/state-versions/sv-wZEoyjPg1KYsjYZg/outputs"
                    }
                }
            },
            "links": {
                "self": "/api/v2/state-versions/sv-wZEoyjPg1KYsjYZg"
            }
        },
        {
            "id": "sv-1ESAifq2ebqrFsd2",
            "type": "state-versions",
            "attributes": {
                "created-at": "2023-12-20T07:04:16.829Z",
                "size": 8520,
                "hosted-state-download-url": "https://app.terraform.io/api/state-versions/sv-1ESAifq2ebqrFsd2/hosted_state",
                "hosted-json-state-download-url": "https://app.terraform.io/api/state-versions/sv-1ESAifq2ebqrFsd2/hosted_json_state",
                "modules": {
                    "root": {
                        "random_string": 1,
                        "aws_db_instance": 1,
                        "aws_db_subnet_group": 1,
                        "data.aws_availability_zones": 1
                    },
                    "root.vpc": {
                        "aws_eip": 2,
                        "aws_vpc": 1,
                        "aws_route": 3,
                        "aws_subnet": 4,
                        "aws_nat_gateway": 2,
                        "aws_route_table": 3,
                        "aws_internet_gateway": 1,
                        "aws_route_table_association": 4
                    },
                    "root.elb-http.elb": {
                        "aws_elb": 1
                    },
                    "root.ec2-instances": {
                        "aws_instance": 4,
                        "data.aws_ami": 1
                    },
                    "root.lb-security-group.sg": {
                        "aws_security_group": 1,
                        "aws_security_group_rule": 6
                    },
                    "root.app-security-group.sg": {
                        "aws_security_group": 1,
                        "aws_security_group_rule": 6
                    },
                    "root.elb-http.elb-attachment": {
                        "aws_elb_attachment": 4
                    }
                },
                "providers": {
                    "provider[\"registry.terraform.io/hashicorp/aws\"]": {
                        "aws_eip": 2,
                        "aws_elb": 1,
                        "aws_vpc": 1,
                        "aws_route": 3,
                        "aws_subnet": 4,
                        "aws_instance": 4,
                        "data.aws_ami": 1,
                        "aws_db_instance": 1,
                        "aws_nat_gateway": 2,
                        "aws_route_table": 3,
                        "aws_elb_attachment": 4,
                        "aws_security_group": 2,
                        "aws_db_subnet_group": 1,
                        "aws_internet_gateway": 1,
                        "aws_security_group_rule": 12,
                        "aws_route_table_association": 4,
                        "data.aws_availability_zones": 1
                    },
                    "provider[\"registry.terraform.io/hashicorp/random\"]": {
                        "random_string": 1
                    }
                },
                "resources": [
                    {
                        "name": "available",
                        "type": "data.aws_availability_zones",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "database",
                        "type": "aws_db_instance",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_db_subnet_group",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "lb_id",
                        "type": "random_string",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/random\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this_name_prefix",
                        "type": "aws_security_group",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "egress_rules",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_rules",
                        "type": "aws_security_group_rule",
                        "count": 4,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_with_self",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "amazon_linux",
                        "type": "data.aws_ami",
                        "count": 1,
                        "module": "root.ec2_instances",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "app",
                        "type": "aws_instance",
                        "count": 4,
                        "module": "root.ec2_instances",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_elb",
                        "count": 1,
                        "module": "root.elb_http.elb",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_elb_attachment",
                        "count": 4,
                        "module": "root.elb_http.elb_attachment",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this_name_prefix",
                        "type": "aws_security_group",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "egress_rules",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_rules",
                        "type": "aws_security_group_rule",
                        "count": 4,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_with_self",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "nat",
                        "type": "aws_eip",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_internet_gateway",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_nat_gateway",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private_nat_gateway",
                        "type": "aws_route",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public_internet_gateway",
                        "type": "aws_route",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_route_table",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_route_table",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_route_table_association",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_route_table_association",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_subnet",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_subnet",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_vpc",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    }
                ],
                "resources-processed": true,
                "serial": 2,
                "state-version": 4,
                "status": "finalized",
                "terraform-version": "1.6.6",
                "vcs-commit-url": null,
                "vcs-commit-sha": null
            },
            "relationships": {
                "run": {
                    "data": {
                        "id": "run-qSU7gjQqAwxgjLaq",
                        "type": "runs"
                    }
                },
                "rollback-state-version": {
                    "data": null
                },
                "created-by": {
                    "data": {
                        "id": "user-VGzYMVaTX6nVYZ3U",
                        "type": "users"
                    },
                    "links": {
                        "self": "/api/v2/users/user-VGzYMVaTX6nVYZ3U",
                        "related": "/api/v2/runs/run-qSU7gjQqAwxgjLaq/created-by"
                    }
                },
                "workspace": {
                    "data": {
                        "id": "ws-pN259iR1J3cW8ivr",
                        "type": "workspaces"
                    }
                },
                "outputs": {
                    "data": [
                        {
                            "id": "wsout-syLHF4vVm1ELesRH",
                            "type": "state-version-outputs"
                        },
                        {
                            "id": "wsout-yCuZCAT1MBrpW9rL",
                            "type": "state-version-outputs"
                        },
                        {
                            "id": "wsout-ujnf5q4ZS3LTEFmL",
                            "type": "state-version-outputs"
                        }
                    ],
                    "links": {
                        "related": "/api/v2/state-versions/sv-1ESAifq2ebqrFsd2/outputs"
                    }
                }
            },
            "links": {
                "self": "/api/v2/state-versions/sv-1ESAifq2ebqrFsd2"
            }
        },
        {
            "id": "sv-WTuTBuwSXsJ7pdar",
            "type": "state-versions",
            "attributes": {
                "created-at": "2023-12-20T07:00:37.618Z",
                "size": 7969,
                "hosted-state-download-url": "https://app.terraform.io/api/state-versions/sv-WTuTBuwSXsJ7pdar/hosted_state",
                "hosted-json-state-download-url": "https://app.terraform.io/api/state-versions/sv-WTuTBuwSXsJ7pdar/hosted_json_state",
                "modules": {
                    "root": {
                        "random_string": 1,
                        "aws_db_instance": 1,
                        "aws_db_subnet_group": 1,
                        "data.aws_availability_zones": 1
                    },
                    "root.vpc": {
                        "aws_eip": 2,
                        "aws_vpc": 1,
                        "aws_route": 3,
                        "aws_subnet": 4,
                        "aws_nat_gateway": 2,
                        "aws_route_table": 3,
                        "aws_internet_gateway": 1,
                        "aws_route_table_association": 4
                    },
                    "root.elb-http.elb": {
                        "aws_elb": 1
                    },
                    "root.ec2-instances": {
                        "aws_instance": 4,
                        "data.aws_ami": 1
                    },
                    "root.lb-security-group.sg": {
                        "aws_security_group": 1,
                        "aws_security_group_rule": 6
                    },
                    "root.app-security-group.sg": {
                        "aws_security_group": 1,
                        "aws_security_group_rule": 6
                    },
                    "root.elb-http.elb-attachment": {
                        "aws_elb_attachment": 4
                    }
                },
                "providers": {
                    "provider[\"registry.terraform.io/hashicorp/aws\"]": {
                        "aws_eip": 2,
                        "aws_elb": 1,
                        "aws_vpc": 1,
                        "aws_route": 3,
                        "aws_subnet": 4,
                        "aws_instance": 4,
                        "data.aws_ami": 1,
                        "aws_db_instance": 1,
                        "aws_nat_gateway": 2,
                        "aws_route_table": 3,
                        "aws_elb_attachment": 4,
                        "aws_security_group": 2,
                        "aws_db_subnet_group": 1,
                        "aws_internet_gateway": 1,
                        "aws_security_group_rule": 12,
                        "aws_route_table_association": 4,
                        "data.aws_availability_zones": 1
                    },
                    "provider[\"registry.terraform.io/hashicorp/random\"]": {
                        "random_string": 1
                    }
                },
                "resources": [
                    {
                        "name": "available",
                        "type": "data.aws_availability_zones",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "database",
                        "type": "aws_db_instance",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_db_subnet_group",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "lb_id",
                        "type": "random_string",
                        "count": 1,
                        "module": "root",
                        "provider": "provider[\"registry.terraform.io/hashicorp/random\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this_name_prefix",
                        "type": "aws_security_group",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "egress_rules",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_rules",
                        "type": "aws_security_group_rule",
                        "count": 4,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_with_self",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.app_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "amazon_linux",
                        "type": "data.aws_ami",
                        "count": 1,
                        "module": "root.ec2_instances",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "app",
                        "type": "aws_instance",
                        "count": 4,
                        "module": "root.ec2_instances",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_elb",
                        "count": 1,
                        "module": "root.elb_http.elb",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_elb_attachment",
                        "count": 4,
                        "module": "root.elb_http.elb_attachment",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this_name_prefix",
                        "type": "aws_security_group",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "egress_rules",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_rules",
                        "type": "aws_security_group_rule",
                        "count": 4,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "ingress_with_self",
                        "type": "aws_security_group_rule",
                        "count": 1,
                        "module": "root.lb_security_group.sg",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "nat",
                        "type": "aws_eip",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_internet_gateway",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_nat_gateway",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private_nat_gateway",
                        "type": "aws_route",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public_internet_gateway",
                        "type": "aws_route",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_route_table",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_route_table",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_route_table_association",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_route_table_association",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "private",
                        "type": "aws_subnet",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "public",
                        "type": "aws_subnet",
                        "count": 2,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    },
                    {
                        "name": "this",
                        "type": "aws_vpc",
                        "count": 1,
                        "module": "root.vpc",
                        "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
                        "index-keys": []
                    }
                ],
                "resources-processed": true,
                "serial": 1,
                "state-version": 4,
                "status": "finalized",
                "terraform-version": "1.6.6",
                "vcs-commit-url": null,
                "vcs-commit-sha": null
            },
            "relationships": {
                "run": {
                    "data": {
                        "id": "run-JTUqTfiuhmconzS8",
                        "type": "runs"
                    }
                },
                "rollback-state-version": {
                    "data": null
                },
                "created-by": {
                    "data": {
                        "id": "user-VGzYMVaTX6nVYZ3U",
                        "type": "users"
                    },
                    "links": {
                        "self": "/api/v2/users/user-VGzYMVaTX6nVYZ3U",
                        "related": "/api/v2/runs/run-JTUqTfiuhmconzS8/created-by"
                    }
                },
                "workspace": {
                    "data": {
                        "id": "ws-pN259iR1J3cW8ivr",
                        "type": "workspaces"
                    }
                },
                "outputs": {
                    "data": [],
                    "links": {
                        "related": "/api/v2/state-versions/sv-WTuTBuwSXsJ7pdar/outputs"
                    }
                }
            },
            "links": {
                "self": "/api/v2/state-versions/sv-WTuTBuwSXsJ7pdar"
            }
        }
    ],
    "links": {
        "self": "https://app.terraform.io/api/v2/state-versions?filter%5Borganization%5D%5Bname%5D=example-org-162af6&filter%5Bworkspace%5D%5Bname%5D=learn-terraform-outputs&page%5Bnumber%5D=1&page%5Bsize%5D=20",
        "first": "https://app.terraform.io/api/v2/state-versions?filter%5Borganization%5D%5Bname%5D=example-org-162af6&filter%5Bworkspace%5D%5Bname%5D=learn-terraform-outputs&page%5Bnumber%5D=1&page%5Bsize%5D=20",
        "prev": null,
        "next": null,
        "last": "https://app.terraform.io/api/v2/state-versions?filter%5Borganization%5D%5Bname%5D=example-org-162af6&filter%5Bworkspace%5D%5Bname%5D=learn-terraform-outputs&page%5Bnumber%5D=1&page%5Bsize%5D=20"
    },
    "meta": {
        "pagination": {
            "current-page": 1,
            "page-size": 20,
            "prev-page": null,
            "next-page": null,
            "total-pages": 1,
            "total-count": 3
        }
    }
}

Reply:
If you are unable to get the outputs with the state version then pls just bundle this data with the state version object
there is no reason for outputs to live without the state version